### PR TITLE
Sends delete XHR when you click delete button

### PR DESCRIPTION
**Pull request by keathleydavidj**
## Purpose
Right now there is  a button for deleting each request on the request index route that does not work.
This PR makes it so that when the user clicks delete for a request it is both removed from the DB and from the UI.

## Approach
pretty simple approach on this one
the requests service now has a method for sending the network request for deleting a record from the database.
An action gets passed down from the route to  the component where the button lives so that it can be triggered on click.

## Learning
One of the things I found myself getting stumped over was Angular template data binding syntax.
for instance, something wrapped in parentheses is responding to an _event_
This is what you'd use for communicating _up_ the tree, from child to parent.
```
<button (click)="doSomeSideEffect()">Side Effect</button>
```

On the other hand, wrapping in square brackets is for passing data _down_ the tree, parent to child.
```
<button [disabled]="disabledState">Button</button>
```
Interestingly, they can be used in conjunction with each other `[(target)]="expression"`

### CODE REVIEW

File: src/app/requests/requests.component.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { RequestsService, Request } from '../requests.service';
+import { analyzeAndValidateNgModules } from '@angular/compiler';
```
Comment (taras) [inline src/app/requests/requests.component.ts]: Do you need this import?

**Review by taras — CHANGES_REQUESTED** [review]
Looking good!

**Comment by taras** [comment]
@keathleydavidj nice job on the PR description. The binding distinction is good to know.

File: src/app/requests/requests.component.ts (hunk 1 above)
Comment (keathleydavidj) [reply src/app/requests/requests.component.ts]: I do not. X]

**Review by taras — APPROVED** [review]

**Pull request merged**